### ISSUE TITLE
[packager] NullPointerException during `jreleaserPackage` for `docker`

### ISSUE THREAD

**Issue by gotson** [jreleaser/jreleaser#1162]
I'm trying to configure JReleaser for Docker, and i stumbled with a `NullPointerException` during the packaging phase, but there's no stacktrace at all, so i can't really understand what's going on.

I've pushed a branch of my repo with the configuration, if that can help understanding what the issue is. JReleaser's configuration is within `komga/build.gradle.kts`.

### Task List

- [x] Steps to reproduce provided
- [x] Stacktrace (if present) provided
- [x] Example that reproduces the problem (link to git repository is ideal)
- [x] Full description of the issue provided (see below)

### Steps to Reproduce

1. git pull https://github.com/gotson/komga/tree/jreleaser
2. `gradle jreleaserPackage`

### Expected Behaviour

I would expect the task to complete somehow, or error on something i can action on (some configuration may still be missing).

### Actual Behaviour

Gradle errors out with a `NullPointerException`

### Environment Information

- **Operating System**: macOS
- **JReleaser Version:** 1.5.0-SNAPSHOT (locally built and published to maven local from 4edf5fb367e20fc22bd243ff8f963fec517110e6)
- **JDK Version:** 11

Please paste stacktraces from `[out|build|target]/jreleaser/trace.log`.

[trace.log](https://github.com/jreleaser/jreleaser/files/10370501/trace.log)

**Comment by gotson** [comment]
I managed to get the stacktrace when running gradle in debug mode.
[trace.log](https://github.com/jreleaser/jreleaser/files/10370513/trace.log)

**Comment by gotson** [comment]
Problem is from https://github.com/gotson/jreleaser/blob/main/core/jreleaser-engine/src/main/java/org/jreleaser/packagers/DockerPackagerProcessor.java#L471-L471

`server` may be null and is not checked.

It would probably make sense to initialize the `server` property of a `Registry` with the same value as the `serverName`.

For example my configuration looked like this:

```kotlin
registries {
        create(DEFAULT_NAME) { externalLogin.set(true) }
        create("ghcr.io") { externalLogin.set(true) }
      }
```

But it needs to look like this:

```kotlin
registries {
        create(DEFAULT_NAME) { externalLogin.set(true) }
        create("ghcr.io") {
          externalLogin.set(true)
          server.set("ghcr.io")
        }
      }
```

I also noticed that when using:

```kotlin
registries {
        create("docker.io") { externalLogin.set(true) }
}
```

Then JReleaser would create 2 configuration entries for the default, and the added one. I don't think that's correct, if the registries are being redefined, the default should not apply.

**Comment by aalmiray** [comment]
Released in v1.5.0 -> https://github.com/jreleaser/jreleaser/releases/tag/v1.5.0
